exec/java-exec/src/test/java/org/apache/drill/exec/fn/impl/TestDateFunctions.java

```diff
@@ -104,10 +104,10 @@ public void testDateDifferenceArithmetic() throws Exception {
 
   @Test
   public void testAge() throws Exception {
-    String[] expectedResults = {"P109M16DT82800S",
-                                "P172M27D",
-                                "P-172M-27D",
-                                "P-39M-18DT-63573S"};
+    String[] expectedResults = {"P107M30DT82800S",
```

Comment:
Thanks for the tests! The code changed four functions, yet we run only one test. Should we have tests for each of the four functions? (The underlying cause of this bug is that we did not have proper tests.)

Also, it is not clear what was wrong with the code: corner cases? Should we test such corner cases here? Diff of two dates across Feb. 29 on a leap year? What other things are wrong with the old approach?

Note that, if we have good coverage, then there is no reason to add a test specific for the one case in DRILL-7926. That case should be just one of many that we include in the overall test of the feature.

Reply:
@paul-rogers 
It seems to me that the main problem was that the month was always considered equal to 30 days.

Reply:
Wow that's some bug! Argh.